SNV and TMB Methods (#66)
AlexsLemonade/OpenPBTA-manuscript: content/03.methods.md

We filtered outputs from both callers on the "PASS" filter, and annotated using The ENSEMBL Variant Effect Predictor [@doi:10/gdz75c], reference release 93, and created MAFs using MSKCC's vcf2maf [@url:https://github.com/mskcc/vcf2maf] v1.6.17.

#### Consensus SNV Calling

> We considered mutations to be the same across callers if they were identical for the following MAF fields: `Chromosome`, `Start_Position`, `Reference_Allele`,  `Allele`, and `Tumor_Sample_Barcode`.

Reviewer: ```suggestion
Our SNV calling process led to separate sets of predicted mutations for each caller.
We considered mutations to describe the same change if they were identical for the following MAF fields: `Chromosome`, `Start_Position`, `Reference_Allele`,  `Allele`, and `Tumor_Sample_Barcode`.
```

> We compared all four callers in their overlap with each other, variant allele frequencies of their calls, base changes called, and genomic regions of their calls [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/compare_snv_callers_plots.Rmd].

Reviewer: ```suggestion
We examined the variant allele frequencies produced by each caller and compared their overlap with each other [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/compare_snv_callers_plots.Rmd].
```
See if this gets the major points of what you were saying with the link to the notebook for details.

> VarDict called numerous variants with low allele frequency [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/vaf_violin_plot.png] in addition to nearly all of the calls identified by the other three callers [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/upset_plot.png].

Reviewer: ```suggestion
VarDict called many variants that were not called by other callers [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/upset_plot.png], while the other callers produced results that were relatively consistent.
Numerous VarDict calls were variants with low allele frequency [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/vaf_violin_plot.png].
```

> Because our filtered set was based on the intersection, we removed it from the consensus calculation.

Reviewer: ```suggestion
We termed mutations shared among the other three callers (Strelka2, Mutect2, and Lancet) to be consensus mutation calls and dropped VarDict due to concerns about its call accuracy in this setting.
In practice because our filtered set was based on the intersection of these three sets and because VarDict called nearly every mutation from the other three callers plus many that were unique to it, the decision to not consider VarDict calls has little impact on the results.
```

> Strelka2 does not call multinucleotide variants (MNV), but instead calls each component SNV as a separate mutation, so we separated MNV calls from Mutect2 and Lancet into consecutive SNVs before comparing them with Strelka2.

Reviewer: I'm confused about why we're doing multinucleotide variants in the SNV section.

Author: The data are all mixed in and different callers call consecutive SNVs differently. I don't know where else this information should go but I'm open to suggestions if there's a better place to put it. I think it is something people should be aware of so I don't necessarily think we should drop it entirely.

Reviewer 2: We have effectively reduced MNVs to consecutive SNVs in the consensus, so I feel like this is the appropriate place to put it. MNVs are distinct from copy number and structural variants, so we won't want to put them in any other place, and I don't think they deserve there own section.

Reviewer 2: On second look, I think it makes sense to move this sentence up to after line 111 ("Our SNV calling process led to..."

> We termed mutations shared among Strelka2, Mutect2, and Lancet consensus mutation calls.

Reviewer: ```suggestion
```

> For downstream analyses, only exonic SNVs (based on Gencode v27 [@url:https://www.gencodegenes.org/human/release_27.html]) are used, to enhance comparability to other studies.

Reviewer: Is this for all downstream analyses or only the TMB ones? Just trying to decide if it goes here or in the TMB section.

Author: I was thinking also for mutational signatures we should stick to exonic so that's why I wrote it this way, but currently that is not just exonic. I'd have to think about this some more though.

> For downstream analyses, only exonic SNVs (based on GENCODE v27 [@url:https://www.gencodegenes.org/human/release_27.html]) are used, to enhance comparability to other studies.

Reviewer: Do you filter any non-exonic SNVs from the consensus called files or do you only filter them in later (TMB / mutational signatures) analyses? This paragraph feels like it probably should go in those sections.

Author: We only filter them based on coding sequences before using them for downstream stuff. The final consensus file has everything.

We filtered outputs from both callers on the "PASS" filter, and annotated using The ENSEMBL Variant Effect Predictor [@doi:10/gdz75c], reference release 93, and created MAFs using MSKCC's vcf2maf [@url:https://github.com/mskcc/vcf2maf] v1.6.17.

#### Consensus SNV Calling

Our SNV calling process led to separate sets of predicted mutations for each caller.
We considered mutations to describe the same change if they were identical for the following MAF fields: `Chromosome`, `Start_Position`, `Reference_Allele`,  `Allele`, and `Tumor_Sample_Barcode`.
We examined the variant allele frequencies produced by each caller and compared their overlap with each other [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/compare_snv_callers_plots.Rmd].

> VarDict called many variants that were not called by other callers [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/upset_plot.png], while the other callers produced results that were relatively consistent.

Reviewer 2: ```suggestion
VarDict calls included many variants that were not identified by other callers [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/upset_plot.png], while the other callers produced results that were relatively consistent with one another.
```

> Numerous VarDict calls were variants with low allele frequency [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/vaf_violin_plot.png].

Reviewer 2: ```suggestion
Many of these VarDict-specific calls were variants with low allele frequency [@url:https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/plots/comparison/vaf_violin_plot.png].
```

> We termed mutations shared among the other three callers (Strelka2, Mutect2, and Lancet) to be consensus mutation calls and dropped VarDict due to concerns about its call accuracy in this setting.

Reviewer 2: Do we want to say accuracy here, or specifically that it was the potentially large number of false positives that we were concerned about?

Author: I've changed this sentence to say false positives instead. I think that's more specifically what we are wanting to refer to here.

> In practice because our filtered set was based on the intersection of these three sets and because VarDict called nearly every mutation from the other three callers plus many that were unique to it, the decision to not consider VarDict calls has little impact on the results.

Reviewer 2: ```suggestion
In practice, because our filtered set was based on the intersection of these three sets and because VarDict called nearly every mutation from the other three callers plus many that were unique to it, the decision to not consider VarDict calls has little impact on the results.
```

Strelka2 does not call multinucleotide variants (MNV), but instead calls each component SNV as a separate mutation, so we separated MNV calls from Mutect2 and Lancet into consecutive SNVs before comparing them with Strelka2.

> For downstream analyses, only coding sequence SNVs (based on GENCODE v27 [@url:https://www.gencodegenes.org/human/release_27.html]) are used, to enhance comparability to other studies.

Reviewer 2: What downstream analyses are we referring to here? I feel like this should perhaps not be so conclusive; some analyses will likely use noncoding sequences as well. Similarly, the definition of effectively surveyed should perhaps be more general. Something like: "the intersection of the genomic ranges considered by the callers used to generate the consensus and where appropriate, the regions of interest, such as coding sequences."

_MYBL1_ [@doi:10.1073/pnas.1300252110], _SNCAIP_ [@doi:10.1038/nature11327], _FOXR2_ [@doi:10.1016/j.cell.2016.01.015], _TTYH1_ [@doi:10.1038/ng.2849], and _TERT_ [@doi:10.1038/ng.3438; @doi:10.1002/gcc.22110; @doi:10.1016/j.canlet.2014.11.057; @doi:10.1007/s11910-017-0722-5] were added to the oncogene list and _BCOR_ [@doi:10.1016/j.cell.2016.01.015] and _QKI_ [@doi:10.1038/ng.3500] were added to the tumor suppressor gene list based on pediatric cancer literature review.
The fusion filtering workflow can be found in the [OpenPBTA Analysis repository](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/fusion_filtering).

> ### Tumor Mutation Burden

Author: Does this mean I should get rid of the other PBTA TMB heading?

> ### Tumor Mutation Burden

Reviewer: I think you should get rid of items in the other heading that are already handled in the SNV section. It looked like this would leave only 1 or 0 sentences, in which case I think bringing that all into this paragraph makes the most sense.

> ### Tumor Mutation Burden

Author: Sounds good. Will do.

We consider tumor mutation burden (TMB) to be the number of consensus SNVs within exons over the portion of the genome *effectively surveyed*.
By *effectively surveyed* we mean the intersection of exons with the ranges considered by the callers used for consensus SNV calls.

Tumor mutation burden here was defined as:

$$ \begin{equation} TMB = (\# \: exonic\: SNVs) \over{(Size\: in\: Mb\: of\: effectively\: surveyed\: genome)} \end{equation} $$

> #### Pediatric Brain Tumor TMB

Author: The elements of this were first defined in #65 but it being all together as the TMB definition is not discussed in that section.

_MYBL1_ [@doi:10.1073/pnas.1300252110], _SNCAIP_ [@doi:10.1038/nature11327], _FOXR2_ [@doi:10.1016/j.cell.2016.01.015], _TTYH1_ [@doi:10.1038/ng.2849], and _TERT_ [@doi:10.1038/ng.3438; @doi:10.1002/gcc.22110; @doi:10.1016/j.canlet.2014.11.057; @doi:10.1007/s11910-017-0722-5] were added to the oncogene list and _BCOR_ [@doi:10.1016/j.cell.2016.01.015] and _QKI_ [@doi:10.1038/ng.3500] were added to the tumor suppressor gene list based on pediatric cancer literature review.
The fusion filtering workflow can be found in the [OpenPBTA Analysis repository](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/fusion_filtering).

### PBTA Tumor Mutation Burden

We consider tumor mutation burden (TMB) to be the number of consensus SNVs per *effectively surveyed* base of the genome.

> $$ \begin{equation} TMB = (\# \: coding sequence\: SNVs) \over{(Size\: in\: Mb\: of\: effectively\: surveyed\: genome)} \end{equation} $$

Reviewer 2: ```suggestion
$$ \textrm{TMB} = \frac{\textrm{\# of coding sequence SNVs}}{\textrm{Size in Mb of {\em effectively surveyed} genome}}
$$
```

Reviewer 2: This is an alternative that is not necessarily better. The {equation} tags add numbering, which I don't know if we are doing in the ms, so either way on that... easy to add or remove later.

We used the total number coding sequence consensus SNVs (GENCODE v27 annotation [@url:https://www.gencodegenes.org/human/release_27.html]) for the numerator and the size of the intersection of the regions considered by Lancet, Strelka2, and Mutect2 with this exome as the denominator.

### TCGA Tumor Mutation Burden

We calculated tumor mutation burden in TCGA using MC3 mutation calls [@doi:10.1016/j.cels.2018.03.002] for TCGA brain-related tumor projects including: LGG (Lower-grade Glioma) [@doi:10.1056/NEJMoa1402121], GBM (Glioblastoma Multiforme) [@doi:10.1016/j.cell.2013.09.034], and PCPG (Pheochromocytoma and Paraganglioma) [@doi:10.1016/j.ccell.2017.01.001].

> The MC3 project provided an exome BED file.

Reviewer 2: Is this exon or CDS?

Author: This one is exon.

_MYBL1_ [@doi:10.1073/pnas.1300252110], _SNCAIP_ [@doi:10.1038/nature11327], _FOXR2_ [@doi:10.1016/j.cell.2016.01.015], _TTYH1_ [@doi:10.1038/ng.2849], and _TERT_ [@doi:10.1038/ng.3438; @doi:10.1002/gcc.22110; @doi:10.1016/j.canlet.2014.11.057; @doi:10.1007/s11910-017-0722-5] were added to the oncogene list and _BCOR_ [@doi:10.1016/j.cell.2016.01.015] and _QKI_ [@doi:10.1038/ng.3500] were added to the tumor suppressor gene list based on pediatric cancer literature review.
The fusion filtering workflow can be found in the [OpenPBTA Analysis repository](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/fusion_filtering).

### PBTA Tumor Mutation Burden

We consider tumor mutation burden (TMB) to be the number of consensus SNVs per *effectively surveyed* base of the genome.

$$ \textrm{TMB} = \frac{\textrm{\# of coding sequence SNVs}}{\textrm{Size in Mb of {\em effectively surveyed} genome}}
$$

> We used the total number coding sequence consensus SNVs (GENCODE v27 annotation [@url:https://www.gencodegenes.org/human/release_27.html]) for the numerator and the size of the intersection of the regions considered by Lancet, Strelka2, and Mutect2 with this exome as the denominator.

Reviewer 2: ```suggestion
We used the total number coding sequence consensus SNVs for the numerator and the size of the intersection of the regions considered by Lancet, Strelka2, and Mutect2 with coding regions (CDS from GENCODE v27 annotation [@url:https://www.gencodegenes.org/human/release_27.html]) as the denominator.
```
